File: awa/sdk_dist/csharp/Awa.Client/Utilities/Activity/CopyFileActivity.cs
using System;
using System.Threading.Tasks;
using Awa.Client.Constants;

namespace Awa.Client.Utilities.Activity
{
    public static partial class ActivityUtilities
    {
        /// <summary>
        /// Copy a file from source to destination.
        /// </summary>
        /// <param name="sourcePath">The source file path.</param>
        /// <param name="destinationPath">The destination file path.</param>
        public static async Task CopyFileActivity(string sourcePath, string destinationPath)
        {
            await Temporalio.Workflows.Workflow.ExecuteActivityAsync(
                AwaConstants.ActivityCopyFile,
                new object?[] { sourcePath, destinationPath },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue,
                    StartToCloseTimeout = TimeSpan.FromSeconds(AwaConstants.FileIoTimeoutSeconds)
                });
        }
    }
}

File: awa/awa/sdk/utils/activity/copy_file_activity.py
from datetime import timedelta
from pathlib import Path

from temporalio import workflow

from awa.sdk import constants


async def copy_file_activity(source_path: str | Path, destination_path: str | Path) -> None:
    """Copy a file from source to destination.

    Args:
        source_path: The source file path.
        destination_path: The destination file path.

    """
    await workflow.execute_activity(
        constants.ACTIVITY_COPY_FILE,
        args=[str(source_path), str(destination_path)],
        task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
        start_to_close_timeout=timedelta(seconds=constants.FILE_IO_TIMEOUT_SECONDS),
    )
